component/async/component.go

```diff
@@ -95,6 +99,15 @@ func (cb *Builder) WithRetries(retries uint) *Builder {
 	return cb
 }
 
+// WithConcurrency specifies the number of worker goroutines for processing messages in parallel
+// default value is '1'
+// do NOT enable concurrency value for in-order consumers, such as Kafka or FIFO SQS
```

Comment:
The problem with this approach is that we actually not have any secure way to disallow this feature to be used with Kafka, FIFO SQS and AMQP. This is exactly the reason why we are thinking about having a separate component in future versions that handles SQS in all of it's glory...
If somebody uses this feature with Kafka the results can be disastrous..

Reply:
You are right, let me find a way around this

Reply:
But what's the problem with AMQP? I think it's perfectly normal to consume out of order. I would just protect Kafka if you agree.

Reply:
AMQP should be working fine i assume except if you need strict ordering.